### PR TITLE
Fix first visit notice

### LINKED ISSUE #217: Can't open hamburger menu on Meta when signed out
https://meta.codidact.com/questions/278713

On Meta, but not other communities, when signed out the hamburger menu does not respond.  I reproduced this problem in Firefox on Android.  I note that Meta is the only community with the new-visitor popup enabled, and I can't dismiss that either.  These might be related but I'm not sure.

### PR DESCRIPTION
**Pull request by sau226**
This PR fixes an issue with the first visit notice, which also resolves an issue with opening the hamburger menu on mobile for sites with such a notice enabled.

This patch was tested on a local dev instance and on first inspection appears to work.

Please mark this as draft or submit a request changes review if changes are required.

GitHub issue linking tag: resolves #217.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -135,8 +135,9 @@ $(document).on('ready', function() {
     });
   });
 
-
-  $('.js-first-visit-notice').on('close.bs.alert', async () => {
-    document.cookie = 'dismiss_fvn=true; path=/; expires=Fri, 31 Dec 9999 23:59:59 GMT';
-  });
+  if (document.cookie.indexOf ('dismiss_fvn') == -1 ) {
```
Comment (ArtOfCode-) [inline app/assets/javascripts/application.js:138]: **suggestion** (code-issues, blocking): `===` for equality checks in JS other than `== undefined`

**nitpick** (code-style, non-blocking): no space before opening paren in function call, i.e. `indexOf('dismiss_fvn')`

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -135,8 +135,9 @@ $(document).on('ready', function() {
...
+    document.getElementById("fvn-dismiss").onclick = function() {
+      document.cookie = 'dismiss_fvn=true; path=/; expires=Fri, 31 Dec 9999 23:59:59 GMT';
+    }
```
Comment (ArtOfCode-) [inline app/assets/javascripts/application.js:141]: **suggestion** (es6, libraries, blocking): we have both ES6 and jQuery available, which we should use to maintain consistency here:

```suggestion
    $('#fvn_dismiss').on('click', () => {
      document.cookie = 'dismiss_fvn=true; path=/; expires=Fri, 31 Dec 9999 23:59:59 GMT';
    });
```

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]
See https://conventionalcomments.org/ for comment format.

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (sau226) [reply app/assets/javascripts/application.js]: All changes in this review comment were implemented

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (sau226) [reply app/assets/javascripts/application.js]: All changes in this comment implemented via committing of the suggested change

**Comment by luap42** [comment]
I see @ArtOfCode- already commented on some stuff. I think that right now using classes and not IDs is more generally consistent, but I’ll have to look, which I’ll be able to do tomorrow evening. Besides that, looks good.

**Pull request merged**